structs/structs.go

```diff
@@ -0,0 +1,58 @@
+package structs
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Result represents the comparison outcome of a field, including its name, old value, and new value.
+type Result struct {
+	FieldName string
+	OldValue  interface{}
+	NewValue  interface{}
+}
+
+/*
+CompareStructs compares two struct instances of the same type
+and returns a list of results with the old and new values of each field tagged with `updateable`.
+*/
+func CompareStructs(old, new interface{}) ([]Result, error) {
+	if reflect.TypeOf(old) != reflect.TypeOf(new) {
+		return nil, fmt.Errorf("both structs must be of the same type")
+	}
+
+	oldValue := reflect.ValueOf(old)
+	newValue := reflect.ValueOf(new)
```

Comment:
what about we also make sure that both old and new are actually structs before calling reflection functions using Kind()?

Reply:
Do we also need to check for possible pointers to structs and dereference them?
```
if oldValue.Kind() == reflect.Ptr {
		oldValue = oldValue.Elem()
	}
```

Reply:
Pointers I am thinking to implement later on. For now just simple types.